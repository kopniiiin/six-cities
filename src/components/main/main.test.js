import React from "react";
import renderer from "react-test-renderer";

import Main from "./main.jsx";

describe(`snapshot test: Main component`, () => {
  it(`should render correctly`, () => {
    const PLACES = [
      `Wood and stone place`,
      `Only wood place`,
      `Only stone place`,
      `Paper place`
    ];

    const placeNameClickHandler = () => {};

    const tree = renderer.create(<Main places={PLACES} placeNameClickHandler={placeNameClickHandler}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
