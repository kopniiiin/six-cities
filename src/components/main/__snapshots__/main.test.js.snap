// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test: Main component should render correctly 1`] = `
<div
  className="page page--gray page--main"
>
  <div />
  <main
    className="page__main page__main--index "
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <CityList
      activeCity="Amsterdam"
      onClick={[Function]}
    />
    <div
      className="cities"
    >
      <div
        className="cities__places-container  container"
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            1
             places to stay in 
            Amsterdam
          </b>
          <WithActiveState
            activeType="popular"
            onTypeChange={[Function]}
          />
          <OfferList
            blockClassName="cities"
            offers={
              Array [
                Object {
                  "city": Object {
                    "location": Object {
                      "coordinates": Array [
                        4,
                        4,
                      ],
                      "zoom": 4,
                    },
                  },
                  "id": "4",
                  "isFavorite": false,
                  "isPremium": false,
                  "location": Object {
                    "coordinates": Array [
                      4,
                      4,
                    ],
                  },
                  "mainPhoto": "photo",
                  "name": "Paper place",
                  "price": 4,
                  "rating": 4,
                  "type": "room",
                },
              ]
            }
            onOfferCardBookmarkButtonClick={[Function]}
            onOfferCardMouseEnter={[Function]}
            onOfferCardMouseLeave={[Function]}
            onOfferCardNameClick={[Function]}
          />
        </section>
        <div
          className="cities__right-section"
        >
          <WithMarkers
            activeMarkerCoordinates={
              Array [
                Array [
                  4,
                  4,
                ],
              ]
            }
            blockClassName="cities"
            centerCoordinates={
              Array [
                4,
                4,
              ]
            }
            markerCoordinates={Array []}
            zoom={4}
          />
        </div>
      </div>
    </div>
  </main>
</div>
`;

exports[`snapshot test: Main component should render message without offers 1`] = `
<div
  className="page page--gray page--main"
>
  <div />
  <main
    className="page__main page__main--index page__main--index-empty"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <CityList
      activeCity="Amsterdam"
      onClick={[Function]}
    />
    <div
      className="cities"
    >
      <div
        className="cities__places-container cities__places-container--empty container"
      >
        <NoOffersMessage
          activeCity="Amsterdam"
        />
        <div
          className="cities__right-section"
        />
      </div>
    </div>
  </main>
</div>
`;
